src/main/java/net/cytonic/cytosis/events/ServerEventListeners.java

```diff
@@ -74,34 +74,23 @@ public static void initServerEvents() {
             final Player player = event.getPlayer();
-            if (CytosisSettings.LOG_PLAYER_CHAT)
-                Cytosis.getDatabaseManager().getMysqlDatabase().addChat(player.getUuid(), event.getMessage());
+            Cytosis.getDatabaseManager().getMysqlDatabase().addChat(player.getUuid(), event.getMessage());
             event.setCancelled(true);
             String originalMessage = event.getMessage();
-            if (!originalMessage.contains("|")) {
-                if (Cytosis.getChatManager().getChannel(player.getUuid()) != ChatChannel.ALL) {
-                    ChatChannel channel = Cytosis.getChatManager().getChannel(player.getUuid());
-                    Component message = Component.text("")
-                            .append(channel.getPrefix())
-                            .append(Cytosis.getRankManager().getPlayerRank(player.getUuid()).orElseThrow().getPrefix())
-                            .appendSpace()
-                            .append(Component.text(player.getUsername(), (Cytosis.getRankManager().getPlayerRank(player.getUuid()).orElseThrow().getTeamColor())))
-                            .append(Component.text(":", Cytosis.getRankManager().getPlayerRank(player.getUuid()).orElseThrow().getChatColor()))
-                            .appendSpace()
-                            .append(Component.text(originalMessage, Cytosis.getRankManager().getPlayerRank(player.getUuid()).orElseThrow().getChatColor()));
-                    Cytosis.getChatManager().sendMessageToChannel(message, Cytosis.getChatManager().getChannel(player.getUuid()));
-                } else {
-                    Component message = Component.text("")
-                            .append(Cytosis.getRankManager().getPlayerRank(player.getUuid()).orElseThrow().getPrefix())
-                            .appendSpace()
-                            .append(Component.text(player.getUsername(), (Cytosis.getRankManager().getPlayerRank(player.getUuid()).orElseThrow().getTeamColor())))
-                            .append(Component.text(":", Cytosis.getRankManager().getPlayerRank(player.getUuid()).orElseThrow().getChatColor()))
-                            .appendSpace()
-                            .append(Component.text(originalMessage, Cytosis.getRankManager().getPlayerRank(player.getUuid()).orElseThrow().getChatColor()));
-                    Cytosis.getOnlinePlayers().forEach((p) -> p.sendMessage(message));
-                }
-            } else player.sendMessage(MM."<red>Hey you cannot do that!");
-            Cytosis.getDatabaseManager().getMysqlDatabase().addChat(player.getUuid(), event.getMessage());
+            ChatChannel channel = Cytosis.getChatManager().getChannel(player.getUuid());
+            switch (channel) {
+                case STAFF, MOD, ADMIN:
+                    if (player.hasPermission(STR."cytonic.chat.\{channel.name()}")) {
```

Comment:
Can you lowercase this? because right now the permission is `cytonic.chat.ADMIN` and I think it should be lowercase